File: Webfuel.Api/Api/Project/ProjectExportApi.cs
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Webfuel.Common;
using Webfuel.Domain;
using Webfuel.Reporting;

namespace Webfuel.App
{
    [ApiService]
    public static class ProjectExportApi
    {
        public static void RegisterEndpoints(IEndpointRouteBuilder app)
        {

            app.MapPut("api/project-export", InitialiseReport)
                .RequireIdentity();
        }

        public static Task<ReportStep> InitialiseReport([FromBody] ProjectExportRequest request, IProjectExportService service)
        {
            return service.InitialiseReport(request);
        }
    }
}

File: Webfuel.App/ClientApp/src/app/api/project-export.api.ts
import { EventEmitter, Injectable, inject } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { ApiService, ApiOptions } from '../core/api.service';
import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from '@angular/router';
import { IDataSource } from 'shared/common/data-source';
import { ProjectExportRequest, ReportStep } from './api.types';

@Injectable()
export class ProjectExportApi {
    constructor(private apiService: ApiService) { }
    
    public initialiseReport (body: ProjectExportRequest, options?: ApiOptions): Observable<ReportStep> {
        return this.apiService.request<ProjectExportRequest, ReportStep>("PUT", "api/project-export", body, options);
    }
}

